src/v/wasm/wasmtime.cc

```diff
@@ -688,18 +696,31 @@ class wasmtime_engine_factory : public factory {
               check_error(error.get());
               check_trap(trap.get());
 
-              return std::make_unique<wasmtime_engine>(
-                _meta.name(),
-                std::move(linker),
-                std::move(store),
-                _module,
-                std::move(xform_module),
-                std::move(sr_module),
-                std::move(wasi_module),
-                std::move(host_function_envs),
-                instance,
-                _alien_thread_pool);
+              // shared_ptr needs to be created in a seastar reactor thread, so
+              // pass back a closure to allocate the engine.
+              return [this,
+                      name = _meta.name(),
+                      linker = std::move(linker),
+                      store = std::move(store),
+                      xform_module = std::move(xform_module),
+                      sr_module = std::move(sr_module),
+                      wasi_module = std::move(wasi_module),
+                      host_function_envs = std::move(host_function_envs),
+                      instance]() mutable {
+                  return ss::make_shared<wasmtime_engine>(
+                    std::move(name),
+                    std::move(linker),
+                    std::move(store),
+                    _module,
+                    std::move(xform_module),
+                    std::move(sr_module),
+                    std::move(wasi_module),
+                    std::move(host_function_envs),
+                    instance,
+                    _alien_thread_pool);
+              };
```

Comment:
😮

Reply:
This will get cleaned up very soon. We'll be able to do this bit without an alien thread in an upcoming PR - this is only a stop gap.

Reply:
its cool as is, just pretty impressive